Documentation/Configuration/TypoScript/Syntax/FileImports/Index.rst

```diff
@@ -0,0 +1,182 @@
+
+@import
+=======
+
+This keyword allows including files inspired by a syntax similar to :code:`SASS`.
+It is restricted, but still allows wildcards on file level. Single files *must* end
+with :file:`.typoscript` if included in frontend Typoscript. In backend TSconfig,
+single files *should* end with :file:`.tsconfig`, but *may* end with
+:typoscript:`.typoscript` as well (for now).
+
+The include logic is a bit more restrictive with TYPO3 v12, previous versions
+have been slightly more relaxed in this regard. See
+:doc:`this Changelog <ext_core:Changelog/12.0/Breaking-97816-TypoScriptSyntaxChanges>`
+for more details.
+
+The following rules apply:
+
+* Multiple files are imported in alphabetical order.
+
+* Recursion is allowed: Imported files can have :typoscript:`@import` statements.
+
+* The :typoscript:`@import` statement does not take a condition clause as the old
+  :typoscript:`<INCLUDE_TYPOSCRIPT condition="">` statement did. That kind of condition
+  should be considered a conceptual mistake. It should not be used.
+
+* .. versionchanged:: 12.0
+
+  It is allowed to put :typoscript:`@import` within a condition. This example imports
+  the additional file only if a frontend user is logged in:
+
+  .. code-block:: typoscript
+
+      [frontend.user.isLoggedIn]
+          @import './userIsLoggedIn.typoscript'
+      [END]
+
+* Both the old syntax :typoscript:`<INCLUDE_TYPOSCRIPT>` and the new one :typoscript:`@import`
+  can be used at the same time.
+
+* Directory imports are not recursive, meaning that a directory import does
+  not automatically travel down its subdirectories.
+
+* Quoting the filename is necessary with the new syntax. Either double quotes
+  (") or single quotes (') can be used.
+
+* Wildcards :typoscript:`*` are only allowed on file level, not on directory level.
+  Only a single wildcard character is allowed.
+
+* Includes relative to the current file location are allowed using :typoscript:`./`
+  as prefix.
+
+* Includes must start with :typoscript:`EXT:` if not relative. Loading files
+  outside of extensions is not possible.
+
+* Directory traversal using :typoscript:`../` is not allowed.
+
+Some examples:
+
+.. code-block:: typoscript
+
+    # Import a single file
+    @import 'EXT:my_extension/Configuration/TypoScript/randomfile.typoscript'
+
+    # Import multiple files in a single directory, sorted by file name
+    @import 'EXT:my_extension/Configuration/TypoScript/*.typoscript'
+
+    # It's possible to omit the file ending. For frontend TypoScript, ".typoscript" is
+    # appended automatically, backend TSconfig allows both ".typoscript" and ".tsconfig"
+    @import 'EXT:my_extension/Configuration/TypoScript/'
+
+    # Import files starting with "foo", ending with ".typoscript" (frontend)
+    @import 'EXT:my_extension/Configuration/TypoScript/foo*'
+
+    # Import files ending with ".setup.typoscript"
+    @import 'EXT:my_extension/Configuration/TypoScript/*setup.typoscript'
```

Comment:
`*setup.typoscript` and `*.setup.typoscript` are different though (see the dot).